test/42-device-api-v2.spec.ts

```diff
@@ -289,12 +289,12 @@ describe('SupervisorAPI [V2 Endpoints]', () => {
 							.body,
 					);
 				});
-			// Deactivate localmode
-			await config.set({ localMode: false });
 		});
 	});
 
-	describe('POST /v2/applications/:appId/start-service', function () {
+	// TODO: setup for this test is wrong, which leads to inconsistent data being passed to
+	// manager methods. A refactor is needed
```

Comment:
What do you mean? Is this an issue with some of the mocked manager methods returning data in an out-of-date format?

Reply:
We could migrate mock service/image setup to use mockerode here

Reply:
There are multiple issues. For one, the mock managers are not setting all the required data, so the assertion in the `inspectByName` was getting triggered (it wasn't getting caught before that). I tried to add it but I kept running with missing pieces for the tests. I think when we work on API parity we should make sure to rewrite some of the API methods to make them more testable, right now we have to stub/mock everything and that is a bad sign.

Reply:
Sounds good, thanks for clarifying.